Improve API Docs in RabbitMQ module (#22568)
ballerina-platform/ballerina-lang: stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/Module.md

> [RabbitMQ](https://www.rabbitmq.com/) is one of the most popular open-source enterprise messaging systems modelled on the Advanced Message Queuing Protocol (AMQP) standard. This guide covers the Ballerina RabbitMQ module and its public API. This module contains the functionality to support messaging with RabbitMQ broker (AMQP 0-9-1). It assumes that the most recent version of Ballerina is used and is built on top of the basics.

Reviewer: ```suggestion
[RabbitMQ](https://www.rabbitmq.com/) is one of the most popular open-source enterprise messaging systems modelled on the Advanced Message Queuing Protocol (AMQP) standard. This guide covers the Ballerina RabbitMQ module and its public API. This module contains the functionality to support messaging with the RabbitMQ broker (AMQP 0-9-1). It assumes that the most recent version of Ballerina is used and is built on top of the basics.
```

### Connecting to RabbitMQ

> The core APIs are `rabbitmq:Connection`, `rabbitmq:Channel`, and `rabbitmq:Message` representing an AMQP 0-9-1 Connection, a Channel, and a Ballerina RabbitMQ Message respectively. The following code connects to a RabbitMQ node using the given parameters (host name, port number, etc.): 

Reviewer: ```suggestion
The core APIs are the `rabbitmq:Connection`, `rabbitmq:Channel`, and `rabbitmq:Message` representing an AMQP 0-9-1 Connection, a Channel, and a Ballerina RabbitMQ Message respectively. The following code connects to a RabbitMQ node using the given parameters (host name, port number, etc.): 
```

                                            password: “guest” });
```

The `rabbitmq:Connection` created above can then be used to open a `rabbitmq:Channel`.

```ballerina
   rabbitmq:Channel newChannel = new(connection);
```

The `rabbitmq:Channel` can now be used to send and receive messages as described in the subsequent sections. 

#### Enabling TLS

> It is possible to encrypt the communication between the Ballerina client and the broker by using TLS-enabled connections. Client and server authentication (peer verification) is also supported. To enable the TLS support in the RabbitMQ broker, the node has to be [configured](https://www.rabbitmq.com/ssl.html#enabling-tls) to know the location of the Certificate Authority bundle, the server's certificate file, and the server's key. A TLS listener should also be configured to know what port to listen to for TLS-enabled client connections.

Reviewer: ```suggestion
It is possible to encrypt the communication between the Ballerina client and the broker by using TLS-enabled connections. Client and server authentication (peer verification) is also supported. To enable the TLS support in the RabbitMQ broker, the node has to be [configured](https://www.rabbitmq.com/ssl.html#enabling-tls) to know the location of the Certificate Authority bundle, the server's certificate file, and the server's key. A TLS listener should also be configured to know which port to listen for TLS-enabled client connections.
```

> Next, the above function calls bind the queue to the exchange with the given routing key. 

Reviewer: ```suggestion
Next, the above function is called to bind the queue to the exchange with the given routing key. 
```

```ballerina
   rabbitmq:Error? deleteResult = newChannel->queueDelete("MyQueue", false, true);
```
- Delete a queue only if it is unused (does not have any consumers):
```ballerina
   rabbitmq:Error? deleteResult = newChannel->queueDelete("MyQueue", true, false);
```
- Delete an exchange:
```ballerina
   rabbitmq:Error? deleteResult = newChannel->exchangeDelete("MyExchange");
```
- Purge a queue (delete all of its messages):
```ballerina
   rabbitmq:Error? purgeResult = newChannel->queuePurge("MyQueue");
```

### Publishing messages

To publish a message to an exchange, use the `basicPublish()` function as follows:

```ballerina
   rabbitmq:Error? sendResult = newChannel->basicPublish("Hello from Ballerina", "MyQueue");
``` 
Setting other properties of the message such as routing headers can be done by using the `BasicProperties` record with the appropriate values. 

### Consuming messages using consumer services

The most efficient way to receive messages is to set up a subscription using a Ballerina RabbitMQ `rabbitmq:Listener` and any number of consumer services. The messages will then be delivered automatically as they arrive rather than having to be explicitly requested. 

> Multiple consumer services can be bound to one Ballerina RabbitMQ `rabbitmq:Listener`. The queue, which the service is listening to is configured in the `rabbitmq:ServiceConfig` annotation of the service. 

Reviewer: ```suggestion
Multiple consumer services can be bound to one Ballerina RabbitMQ `rabbitmq:Listener`. The queue to which the service is listening is configured in the `rabbitmq:ServiceConfig` annotation of the service. 
```

> >**Note:** The default thread pool size used in Ballerina is number of processors available * 2. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.

Reviewer: ```suggestion
>**Note:** The default thread pool size used in Ballerina is the number of processors available * 2. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
```
